File: MeetUpUI/src/app/dtos/user.dto.ts
export class UserDto{
    username!: string;
    id!: string;
    isInCall!: boolean;
    roomId?: string | null;

    constructor(username: string, id:string, isInCall: boolean = false, roomId: string | null = null){
        this.username = username;
        this.id = id;
        this.isInCall = isInCall;
        this.roomId = roomId;
    }
}

File: MeetUpApi/MeetUp.Api/Dtos/UserDto.cs
namespace Nexcall.Api.Dtos
{
    public class UserDto
    {
        public string AppUserId { get; set; }
        public string Email { get; set; }
        public string Username { get; set; }
        public string Id { get; set; }
        public bool IsInCall { get; set; }
        public string? RoomId { get; set; }

        public UserDto(string id, string appUserId, string username, string email)
        {
            this.Username = username;
            this.Id = id;
            this.AppUserId = appUserId;
            this.Email = email;
            this.IsInCall = false;
            this.RoomId = null;
        }

        public UserDto(string id, string appUserId, string username, string email, bool isInCall, string? roomId)
        {
            this.Username = username;
            this.Id = id;
            this.AppUserId = appUserId;
            this.Email = email;
            this.IsInCall = isInCall;
            this.RoomId = roomId;
        }
    }
}
